src/Uno.UWP/UI/Notifications/ToastNotifier.Android.cs

```diff
@@ -0,0 +1,326 @@
+			}
+			toasttext = resName; 
+			var retVal = Windows.ApplicationModel.Resources.ResourceLoader.GetForCurrentView()?.GetString(toasttext);
+			if (retVal is null)
+			{   // we have no such string
+				return toasttext;
+			}
+			return retVal;
+		}
+
+		private Android.App.Notification.Builder GetToastBuilder()
+		{
+			// if API < 11, then another version...
+			if (Android.OS.Build.VERSION.SdkInt >= Android.OS.BuildVersionCodes.O)
+			{
+				return new Android.App.Notification.Builder(Android.App.Application.Context, _channelID);
+			}
+
+#pragma warning disable CS0618 // Type or member is obsolete - compilation is for newer SDK, but runtime is on older SDK
+			return new Android.App.Notification.Builder(Android.App.Application.Context);
+#pragma warning restore CS0618
+
+		}
+
+		private int GetIconId(string packageName)
+		{
+			var info = Android.App.Application.Context?.ApplicationContext?.PackageManager?.GetApplicationInfo(packageName, 0);
+
+			int iconId;
+			if (info != null)
+			{
+				iconId = info.Icon;
+			}
+			else
+			{
+				// try another method - using more resources (CPU and memory), as Intent is involved
+				var context = Android.App.Application.Context;
+				if (context?.PackageName is null)
+				{
+					throw new InvalidOperationException("ToastNotifier:GetIconId, Context.PackageName is null");
+				}
+
+				var intent = context.PackageManager?.GetLaunchIntentForPackage(context.PackageName);
+				if (intent is null)
+				{
+					throw new InvalidOperationException("ToastNotifier:GetIconId, GetLaunchIntentForPackage is null");
+				}
+
+				var resolveInfo = context.PackageManager?.ResolveActivity(intent, Android.Content.PM.PackageInfoFlags.MatchDefaultOnly);
+				if (resolveInfo is null)
+				{
+					throw new InvalidOperationException("ToastNotifier:GetIconId, ResolveActivity is null");
+				}
+				iconId = resolveInfo.IconResource;
+			}
+
+			// iconId == null means error, but we throw exception in Show method - logic would be more readable
+			return iconId;
+
+		}
+
+		private void SetToastTexts(Android.App.Notification.Builder builder, XmlDocument xmlDoc)
+		{
+			string toastTitleText = "";
+			StringBuilder toastText = new StringBuilder("");
+
+			var childToast = xmlDoc.GetElementsByTagName("text");
+			if ((childToast != null) && childToast.Count > 0)
+			{
+				// first text - bigger text (title)
+				toastTitleText = ConvertToastTextToString(childToast![0]!.InnerText);
+
+				if (childToast.Count > 1)
+				{
+					toastText.Append(ConvertToastTextToString(childToast![1]!.InnerText));
+
+					for (int childIndex = 2; childIndex < childToast.Count; childIndex++)
+					{   // in most scenarios, this loop will never iterate
+						// separate lines with space and \n: \n as line splitting for newer Android, space - for older
+						toastText.Append(" \n" + ConvertToastTextToString(childToast![childIndex]!.InnerText!));

```

Comment:
Same here for nullable exclusions.

Reply:
Same as previous.
Inside loop which tests existence of element (`childIndex < childToast.Count`), this element has to exist.

Reply:
Ok, adding one more test.